ecosystem/sep-0039.md

```diff
@@ -0,0 +1,182 @@
+
+
+## Motivation
+With NFT marketplaces on the rise in the Stellar network (e.g. [Litemint](https://litemint.com/) and [Stellar Quest](https://quest.stellar.org/)), it's important to establish standards to ensure cross-compatibility of NFTs. Certain platforms may choose to store their asset---whether that be an image as we often see today or any arbitrary digital trinket---on the Stellar ledger itself rather than delegating to an off-chain storage mechanism such as [IPFS](https://ipfs.io/).
+
+A standard storage mechanism will help these marketplaces and wallets render or otherwise represent the NFT within their applications, enhancing their transferability.
+
+
+## Abstract
+We present an efficient encoding mechanism to store arbitrary binary data on the Stellar ledger via the `ManageData` operation, specifically catering towards NFTs. It will concatenate a special SEP-39 header alongside a [Media Type](https://www.iana.org/assignments/media-types/media-types.xhtml) containing asset metadata alongside the aforementioned encoded-text as keys and raw binary data as values.
+
+
+## Specification
+First, we will discuss a common NFT issuance model and its ownership implications. This is not a required model, but sets the stage for why a standardized encoding of digital assets is necessary.
+
+**Warning**: Because of how indexing works (that is, each data entry for the asset begins with an index), an account can only hold a single SEP-39 asset at a time. Users should also take care not to have colliding keys that may interfere with ordering, and should lean on the (optional) checksum parameter if they're worried about decoding inconsistencies.
+
+### NFT Issuance
+Given an arbitrary digital asset, here's how issuance typically works:
+
+  1. An account is created to represent the digital asset on the ledger
+  2. The account stores the entire digital asset within its managed data entries.
+  3. The account issues a Stellar asset representing ownership over the digital asset we just stored.
+  4. Any account holding that Stellar asset lays claim to the digital asset.
+
+Notice that ownership of the token directly conveys ownership of a unit of the asset, entirely within the Stellar ledger. Other scenarios such as locking down the issuing account to avoid supply increases, ownership transfers via payments, etc. are built on top of this basic model and are thus extraneous to this SEP.
+
+Note also that this SEP does not address concerns around *immutability*. While there is an optional checksum component (addressed later), the account itself is free to modify the data at-will if it is not locked.
+
+Our exclusive focus in this SEP is Step 2: **storing the asset within the account.**
+
+### Data Format
+We begin by describing the raw data format, then dive into each component.
+
+    00|version|metadata length|media-type;param1=value;param2=value;etc<binary data>
```

Comment:
Are we printing this in raw string format vs encoding it along with the rest of the `<binary data>`? Seems like we should just concatenate the entire thing as a singular buffer. And only have the index maybe be the `radix of 36 index|version|metadata length|<metadata binary><fields binary>`. Honestly I think the entire sep39 format should be one super buffer with no consideration for any raw string format. I could be convinced the indexing could be the radix of 36, that's kinda clever and helps both sorting and avoiding clashes with non sep39 key:value pairs.

Reply:
```suggestion
    00<1 byte version buffer><2 byte metadata length buffer><metadata buffer><concatenated buffer of field values>
    e.g. 00|1|30|<text/plain;l=10;k=title,text/plain;l=50;k=description,image/png;l=500;k=image>|<concatenated buffer of field values>
```

Reply:
I was thinking more about the radix thing, and honestly _all_ we need from it is sortability from Horizon. It offers no other value. The decoder can ignore it entirely. With that in mind, we can even use `aa`, `ab`, ... through `zz`.

Reply:
Let's stick with numbers. 00-9z is less likely to clash with a non sep39 key than strictly alphanumeric chars